<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect><x>0</x><y>0</y><width>800</width><height>480</height></rect>
  </property>
  <property name="windowTitle"><string>Face Doorlock</string></property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="mainVerticalLayout">
    <item>
     <layout class="QHBoxLayout" name="mainContentLayout">
      <!-- LEFT: Camera preview (unchanged name) --> 
      <item>
       <widget class="QLabel" name="cameraLabel">
        <property name="minimumSize"><size><width>640</width><height>480</height></size></property>
        <property name="scaledContents"><bool>true</bool></property>
        <property name="text"><string/></property>
       </widget>
      </item>
      <!-- RIGHT: Tabs (keep existing controls under the first tab) -->
      <item>
       <widget class="QTabWidget" name="rightTabWidget">
        <property name="tabPosition"><enum>QTabWidget::North</enum></property>
        <property name="tabShape"><enum>QTabWidget::Rounded</enum></property>

        <!-- TAB 1: Control (contains the original rightPanelLayout contents) -->
        <widget class="QWidget" name="tabControl">
         <attribute name="title"><string>Control</string></attribute>
         <layout class="QVBoxLayout" name="rightPanelLayout">
		  <item>
			  <layout class="QHBoxLayout" name="ledBarLayout">
			  	<property name="spacing"><number>8</number></property>
				<!-- BLE -->
				<item>
					<widget class="QLabel" name="labelBle">
						<property name="text"><string>BLE</string></property>
						<property name="alignment"><set>Qt::AlignVCenter|Qt::AlignLeft</set></property>
					</widget>
				</item>
				<item>
					<!-- Promoted Widget:LedWidget -->
					<widget class="LedWidget" name="bleLed">
						<property name="minimumSize">
							<size><width>18</width><height>18</height></size>
						</property>
						<property name="maximumSize">
							<size><width>22</width><height>22</height></size>
						</property>
					</widget>
				</item>

				<!-- 중간 여백 -->
				<item>
					<spacer name="hSpacerBetween">
						<property name="orientation"><enum>Qt::Horizontal</enum></property>
						<property name="sizeHint"><size><width>16</width><height>20</height></size></property>
					</spacer>
				</item>
				<!-- DOOR -->
				<item>
					<widget class="QLabel" name="labelDoor">
						<property name="text"><string>DOOR</string></property>
						<property name="alignment"><set>Qt::AlignVCenter|Qt::AlignLeft</set></property>
					</widget>
				</item>
				<item>
					<!-- Promoted Widget: LedWidget -->
					<widget class="LedWidget" name="doorLed">
						<property name="minimumSize">
							<size><width>18</width><height>22</height></size>
						</property>
					</widget>
				</item>
				<!-- 오른쪽 여유 공간 -->
				<item>
					<spacer name="hSpacerStretch">
						<property name="orientation"><enum>Qt::Horizontal</enum></property>
						<property name="sizeHint"><size><width>40</width><height>20</height></size></property>
					</spacer>
				</item>
			  </layout>
		  </item>
          <item>
           <widget class="QPushButton" name="registerButton">
            <property name="text"><string>등록 시작</string></property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="resetButton">
            <property name="text"><string>사용자 초기화</string></property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="showUsersList">
            <property name="text"><string>사용자 목록</string></property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="showUserImages">
            <property name="text"><string>사용자 이미지 목록</string></property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="ExitButton">
            <property name="text"><string>종료</string></property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="recognitionLabel">
            <!-- Set a friendly default text; keep the original object name -->
						<property name="pixmap"><pixmap>/root/trucnk/faceRecognizer_doorlock/assert/images/gardix_logo.png</pixmap></property>
            <property name="alignment"><set>Qt::AlignCenter</set></property>
            <property name="styleSheet"><string notr="true">font-size: 16px; color: #3498db; font-weight: bold;</string></property>
           </widget>
          </item>
         </layout>
        </widget>

        <!-- TAB 2: Settings (hosts the SettingsTab.ui you already have) -->
        <widget class="QWidget" name="tabSettings">
         <attribute name="title"><string>Settings</string></attribute>
         <layout class="QVBoxLayout" name="settingsPageLayout">
          <!-- Option A (Designer): Promote this placeholder to SettingsTab (class: SettingsTab, header: SettingsTab.ui or a wrapper header) -->
          <!-- Option B (Code): Find this container by objectName and add a loaded UI into it at runtime -->
          <item>
           <widget class="QGroupBox" name="groupControlActions">
            <property name="title"><string>장비 제어</string></property>
            <layout class="QVBoxLayout" name="controlActionsLayout">

             <item>
              <layout class="QHBoxLayout" name="rowActions1">
               <item>
                <widget class="QPushButton" name="btnRefresh">
                 <property name="text"><string>새로고침</string></property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="btnRestartCamera">
                 <property name="text"><string>카메라 재시작</string></property>
                </widget>
               </item>
              </layout>
             </item>

             <item>
              <layout class="QHBoxLayout" name="rowActions2">
               <item>
                <widget class="QPushButton" name="btnUnlockDoor">
                 <property name="text"><string>도어 열기</string></property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="btnLockDoor">
                 <property name="text"><string>도어 잠그기</string></property>
                </widget>
               </item>
              </layout>
             </item>

             <item>
              <layout class="QHBoxLayout" name="rowActions3">
               <item>
                <widget class="QPushButton" name="btnRetrain">
                 <property name="text"><string>인식기 재학습</string></property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="btnExportLogs">
                 <property name="text"><string>로그 내보내기</string></property>
                </widget>
               </item>
              </layout>
             </item>

             <item>
              <widget class="QLabel" name="actionStatusLabel">
               <property name="text"><string>Ready</string></property>
               <property name="wordWrap"><bool>true</bool></property>
               <property name="styleSheet">
                <string notr="true">color: #666; font-size: 12px;</string>
               </property>
              </widget>
             </item>

            </layout>
           </widget>
          </item>
          <item>
           <widget class="QWidget" name="settingsTabContainer">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
         </layout>
        </widget>

       </widget>
      </item>

     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
   <customwidgets>
    <customwidget>
      <class>LedWidget</class>          <!-- ✅ 클래스명 -->
      <extends>QWidget</extends>
      <header>LedWidget.h</header>      <!-- ✅ 정확한 헤더명(대소문자 일치) -->
    </customwidget>
  </customwidgets>
 <resources/>
 <connections/>
</ui>
